app/helpers/application_helper.rb

```diff
@@ -1,4 +1,19 @@
 module ApplicationHelper
+  APP_HOST = ENV['APP_HOST']
+  APP_HOST_LEGACY = ENV['APP_HOST_LEGACY']
+
+  def app_host_legacy?(request)
+    return false if APP_HOST_LEGACY.blank?
+
+    Regexp.new(APP_HOST_LEGACY).match?(request.base_url)
+  end
+
+  def auto_switch_domain?(request, user_signed_in)
+    maybe_switch_domain_enabled = request.params.key?(:switch_domain) || Flipper.enabled?(:switch_domain)
+
+    maybe_switch_domain_enabled && !user_signed_in && app_host_legacy?(request)
```

Comment:
pas `user_signed_in?` ? (avec le ?)
et je me rappelle plus pourquoi on veut le fairei que n'étant pas connecté, on en a du en parler mais j'ai zappé. En tout cas ok peut se limiter à ça au début

Reply:
si tu peux rajouter une méthode publique
```ruby
def switch_domain_enabled? 
  request.params.key?(:switch_domain) || Flipper.enabled?(:switch_domain)
end
```

comme ça je reprend la même pour l'affichage de la bannière

Reply:
`user_signed_in?`, c'est un helper devise, pas dispo ds les specs des helpers. bref, d'ordre general, je prefere ne pas me reposer sur trop de dependance ds la methode, mais les donner

Reply:
pour la question du prk en `user_signed_in?`, je crois qu'on voulait faire de l'opt-in, sinon on aurait redirigé automatiquement via le token.

Reply:
je crois qu'on parlait de l'optin si on ne faisait pas de détection auto. De toute façon je suis plutôt de voir ce que donne en vrai la détection, et si c'est OK assez rapidement migrer tout le monde quand elle est OK